bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -198,10 +200,19 @@ public void stateUpdated(Item item, State state) {
                 event.valuecolor = itemUIRegistry.getValueColor(w);
                 event.widgetId = itemUIRegistry.getWidgetId(w);
                 event.visibility = itemUIRegistry.getVisiblity(w);
+                // event.item contains data from the item including its state (in event.item.state)
                 event.item = EnrichedItemDTOMapper.map(item, false, null, null);
 
-                // adjust the state according to the widget type
-                event.item.state = itemUIRegistry.getState(w).toFullString();
+                // event.state is an adjustment of the item state to the widget type.
+                // In case this state is big and identical to the item state, an indirection to the
+                // item state is provided by setting its value to "#item.state#" in order to avoid
```

Comment:
This looks like a very nasty hack to me. How is any client be supposed to know this? If a String item state changes by one character from 12 to 13 in length, the events will all of a sudden have a different content?

Those magic strings should imho be really avoided.

It would probably be better to simply say that the widget state is generally only provided if it differs from the item state. Clients would not have to check for any magic strings, but could simply implement a simple fallback to the item state. Or am I oversimplifying things here?

Reply:
Yes I like very much this idea. I don't remember why I abandoned it :)
I think I already added in the past the transformed state in the item data which is optional, that is present only if different from the state. So it is clearly doable.

The only problem I see with that solution is that a client UI will not be able to distinguish old behaviour from new behaviour just with the event. It will probably require to check the API version.

Reply:
I think I abandoned this idea due to @digitaldan comment: https://github.com/eclipse/smarthome/issues/4445#issuecomment-341996820

Reply:
But as a logic is required by the way if we don't want to duplicate big data, I think this argument is no more valid.